clang/test/SemaCXX/attr-gsl-owner-pointer.cpp

```diff
@@ -0,0 +1,36 @@
+// RUN: %clang_cc1 -fsyntax-only -verify %s
+
+int [[gsl::Owner]] i;
+//expected-error@-1 {{'Owner' attribute cannot be applied to types}}
+void [[gsl::Owner]] f();
+//expected-error@-1 {{'Owner' attribute cannot be applied to types}}
+
+struct S {
+};
+
+S [[gsl::Owner]] Instance;
+//expected-error@-1 {{'Owner' attribute cannot be applied to types}}
+
+class [[gsl::Owner]] OwnerMissingParameter {};
+//expected-error@-1 {{'Owner' attribute takes one argument}}
+class [[gsl::Pointer]] PointerMissingParameter {};
+//expected-error@-1 {{'Pointer' attribute takes one argument}}
+
+
+class [[gsl::Owner(7)]] OwnerDerefNoType {};
+class [[gsl::Pointer("int")]] PointerDerefNoType {};
+
+
+class [[gsl::Owner(int)]] [[gsl::Pointer(int)]] BothOwnerPointer {};
```

Comment:
I think we should diagnose this as an error. A TODO is also fine if we plan to do it in a later commit.